### ISSUE TITLE
History for structured product labels 

### ISSUE THREAD

**Issue by dportnoy** [demand-driven-open-data/ddod-intake#25]
**Requester:** @rxdata
**Source:** Phone call with Lawrence at rxData on 4/2/2015
**Notes from call:**
- OpenFDA lacks historical product labels / indications
- To access historical label revisions, rxData currently relies on Structured Product Labeling (SPL) public use files.  
- They may also look at XML format files at AccessData.fda.gov, specifically: http://www.accessdata.fda.gov/spl/data/[SPL_document_id].xml   (See [SPL
  Implementation Guide](http://www.fda.gov/downloads/ForIndustry/DataStandards/StructuredProductLabeling/UCM321876.pdf))
- Another limitation is that OpenFDA API response is missing HTML formatting markers, such as line breaks

**Comment by RxData** [comment]
Use case summary
• Title: History for structured product labels
• Work item: https://github.com/demand-driven-open-data/ddod-intake/issues/#
• Activity log: DDOD activity log for use case
• Solution: https://github.com/demand-driven-open-data/ddod-intake/wiki/Solution-#
Description
• OpenFDA API lacks historical product labels / indications
• To access historical label revisions, RxData currently relies on Structured Product Labeling (SPL) public use XML files OR AccessFDA database pdf links
• Another limitation is that OpenFDA API response is missing HTML formatting markers, such as line breaks
Value
• Value to customer: Looking at pharmaceutical market access in context of changing labeling information
• Value to industry/public: Understand labeling changes
Customer
• Pharmaceutical companies
Current data and limitation
• Data source: SPL XMLs, AccessFDA database, OpenFDA Label API 
• Challenge: Each source has only part of the information needed
Specifications
• Fields: 
o   SPL Set ID
o   PDF link
o   BLA number and supplement number
• Update frequency: Weekly/daily
• Joins between datasets: SPL XMLs, AccessFDA database, OpenFDA Label API

**Comment by dportnoy** [comment]
Created specifications document from notes: ~~https://github.com/demand-driven-open-data/ddod-intake/wiki/Use-Case-25~~

**Update:** Moved page for full use case specifications and solution: http://hhs.ddod.us/wiki/Use_Case_25

**Comment by dportnoy** [comment]
@RxData  Just to confirm, you're looking for SPL for drug only, not devices or biologics.

**Comment by RxData** [comment]
Yes, we are only interested in prescription drugs.

On Wed, Jun 17, 2015 at 11:25 AM, David X Portnoy notifications@github.com
wrote:

> @RxData https://github.com/RxData Just to confirm, you're looking for
> SPL for drug only, not devices or biologics.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/demand-driven-open-data/ddod-intake/issues/25#issuecomment-112844846
> .

## 

Joe Segal

Co-Founder & CTO
_RxData, Inc._
447 Broadway, 2nd Floor
New York, NY 10013
phone: +1 (321) 356-6305
jsegal@rxdata.net | rxdata.net

**Comment by ftrotter** [comment]
I think this is important for over-the-counter medications as well. Not just prescription.

This is a great idea, BTW.

-FT

**Comment by ghost** [comment]
@RxData I'd like to help with this issue.  Can you say more about the use cases around changes in labeling information?  I have my own from an NLM Pillbox perspective, but want to understand yours.

@ftrotter +1 to OTCs.  A solution for RX labels will most likely be useable for OTC and homeopathic products, with the limitation that their participation in SPL is substantial lower, but growing.

**Comment by ghost** [comment]
@RxData these calls to the DailyMed may be useful.  I'm not on the DailyMed team, but would be happy to take feedback to them.

Returns the version history for a given Set ID in either json or xml.
http://dailymed.nlm.nih.gov/dailymed/services/v2/spls/{SetID}/history.json
http://dailymed.nlm.nih.gov/dailymed/services/v2/spls/{SetID}/history.xml
example:
http://dailymed.nlm.nih.gov/dailymed/services/v2/spls/4724dbb4-3613-4e6a-948f-a43d34f97f06/history.json

Return the specific version of a label.  This will return a .zip file containing the label xml and all images and support files associated with that label.  This is a new function and isn't yet documented on DailyMed.
http://dailymed.nlm.nih.gov/dailymed/getFile.cfm?type=zip&setid={SetID}&version={version number}
example:
http://dailymed.nlm.nih.gov/dailymed/getFile.cfm?type=zip&setid=4724dbb4-3613-4e6a-948f-a43d34f97f06&version=13

Return all previous versions of the label in a .zip file.  Each previous version is in a separate enclosed .zip file.
http://dailymed.nlm.nih.gov/dailymed/services/v2/spls/{SetID}/history.zip
example:
http://dailymed.nlm.nih.gov/dailymed/services/v2/spls/4724dbb4-3613-4e6a-948f-a43d34f97f06/history.zip

**Comment by dportnoy** [comment]
@lostonroute66  Could you summarize the advantages DailyMed would have over the public use file (http://www.accessdata.fda.gov/spl/data/[SPL_document_id].xml) option?   (Or do you need to hear back from @RxData on additional specifics first?)

**Comment by ghost** [comment]
@DavidXPortnoy problem with using that option is you need to know the filename [SPL_document_id](id root> in the label) which is different than the Set ID.  Each new version of a label has the same Set ID, but a new filename. The DailyMed history call does not return the file name of the previous versions - only the version number and published date.

A work around would be to look inside the .zip files from my previous post to get the filenames.  But at that point you would already have the file.

The more elegant solution would be for either DailyMed to return the filenames for each previous version with the history call or for OpenFDA to develop another solution.  I'll ask the DailyMed team about this.

btw the syntax I have to use for the call in your post is http://www.accessdata.fda.gov/spl/data/[SPL_document_id]/[SPL_document_id].xml

**Comment by ghost** [comment]
If it works with how you're organizing these, I would recommend splitting this use case into two.

1) Access previous versions of label xml files (not specifying a source, unless there's a reason)
2) OpenFDA missing HTML formatting markers, such as line breaks

**Comment by dportnoy** [comment]
Go for it

**Comment by ghost** [comment]
@RxData do you want to be able to specify the version for which you want an xml file?  Or do you want to give a Set ID and get xml files for every previous version (there can be many in some cases)?

**Comment by RxData** [comment]
@lostonroute66 (and @DavidXPortnoy ), thank you very much for your comments - very much appreciated. 

Agree to the split of:

1) Access previous versions of label xml files (not specifying a source, unless there's a reason)
2) OpenFDA missing HTML formatting markers, such as line breaks

For our purposes, your answers are sufficient for 1) and very valuable (so thanks again!) - so you may close this topic unless anyone else could benefit of course.

Regarding 2), this would be very useful especially regarding line breaks. And, sometimes both versions of Indications and Usage are within the same field:

There's a version of the indication in the HIGHLIGHTS OF PRESCRIBING INFORMATION (https://www.evernote.com/l/AOwdBI5CClFLr4uwUzYxT3vwdlqtVhnRf_k in green)...
And one in the FULL PRESCRIBING INFORMATION (https://www.evernote.com/l/AOxNpqVYXfZLiLCsPPc_Ss1laD_Gbx1xebc in blue)
--> They are both merged together (sometimes) when it comes to the API returning the Indication.. For information, the example above was retrieved from http://www.accessdata.fda.gov/drugsatfda_docs/label/2011/125377s0000lbl.pdf

Thanks again!

**Comment by ghost** [comment]
@RXData is this last part (Yervoy indications and usage) related to part 2 (line breaks)?

**Comment by ghost** [comment]
@RxData are you familiar with http://www.fda.gov/Drugs/GuidanceComplianceRegulatoryInformation/LawsActsandRules/ucm084189.htm ?  Especially the "The New Look" section.

On the data side, when a label has Highlights, there will be two Indications and Usage sections.  One in the full label, one in the highlights. Each will have different content as described in the above link.

Can you post and API (OpenFDA?) link?  I'm not sure how OpenFDA handles Highlights.

Thanks

**Comment by dportnoy** [comment]
@RxData  On formatting issues.  Could we recap please.  Looks like there's an open question from @lostonroute66.  It would be good to understand if you're satisfied with where the formatting of the use case ended up or if you have more challenges to address.

**Comment by dportnoy** [comment]
### Splitting **formatting** into separate use case #45

...so we can manage it better.

Knowledge base: http://hhs.ddod.us/wiki/Use_Case_45

**Comment by RxData** [comment]
@DavidXPortnoy @lostonroute66 Thanks both for your help! We are now actually all good for both formatting and history. Thanks again very much!

**Comment by ghost** [comment]
Procedure for accessing previous versions of labels via DailyMed API has been added to DDOD wiki, [Use Case 25](http://hhs.ddod.us/wiki/Use_Case_25:_History_for_structured_product_labels).

Thank you @RxData and @dportnoy 

@DavidXPortnoy could you please close this issue?  thx
